openflow15/multipart.go

```diff
@@ -0,0 +1,3594 @@
+			return
+		}
+		data = append(data, b...)
+	}
+
+	log.Debugf("Sending MultipartRequest (%d): %v", len(data), data)
+
+	return
+}
+
+func (s *MultipartRequest) UnmarshalBinary(data []byte) error {
+	err := s.Header.UnmarshalBinary(data)
+	n := s.Header.Len()
+
+	s.Type = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	s.Flags = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	n += 4 // for padding
+	for n < s.Header.Length {
+		var req util.Message
+		switch s.Type {
+		case MultipartType_Desc:
+			// The request body is empty.
+		case MultipartType_FlowDesc:
+			// The request body is struct ofp_flow_stats_request.
+			req = NewFlowStatsRequest()
+		case MultipartType_AggregateStats:
+			// The request body is struct ofp_aggregate_stats_request.
+			req = new(AggregateStatsRequest)
+		case MultipartType_TableStats:
+			// The request body is empty.
+		case MultipartType_Port:
+			// The request body is struct ofp_port_multipart_request.
+			req = new(PortMultipartRequst)
+		case MultipartType_QueueStats:
+			// The request body is struct ofp_queue_multipart_request.
+			req = new(QueueMultipartRequest)
+		case MultipartType_GroupStats:
+			// The request body is struct ofp_group_multipart_request.
+			req = new(GroupMultipartRequest)
+		case MultipartType_GroupDesc:
+			// The request body is struct ofp_group_multipart_request.
+			req = new(GroupMultipartRequest)
+		case MultipartType_GroupFeatures:
+			// The request body is empty.
+		case MultipartType_MeterStats:
+			// The request body is struct ofp_meter_multipart_request.
+			req = new(MeterMultipartRequest)
+		case MultipartType_MeterDesc:
+			// The request body is struct ofp_meter_multipart_request.
+			req = new(MeterMultipartRequest)
+		case MultipartType_MeterFeatures:
+			// The request body is empty.
+		case MultipartType_TableFeatures:
+			// The request body is either empty or contains an array of
+			// struct ofp_table_features containing the controller’s
+			// desired view of the switch. If the switch is unable to
+			// set the specified view an error is returned.
+			req = new(TableFeatures)
+		case MultipartType_PortDesc:
+			// The request body is struct ofp_port_multipart_request.
+			req = new(PortMultipartRequest)
+		case MultipartType_TableDesc:
+			// The request body is empty.
+		case MultipartType_QueueDesc:
+			// The request body is struct ofp_queue_multipart_request.
+			req = new(QueueMultipartRequest)
+		case MultipartType_FlowMonitor:
+			// The request body is an array of struct ofp_flow_monitor_request.
+			req = new(FlowMonitorRequest)
+		case MultipartType_FlowStats:
+			//  The request body is struct ofp_flow_stats_request.
+			req = NewFlowStatsRequest()
+		case MultipartType_ControllerStatus:
+			// The request body is empty.
+		case MultipartType_BundleFeatures:
+			// The request body is ofp_bundle_features_request.
+			req = new(BundleFeaturesRequest)
+		case MultipartType_Experimenter:
+		}
+
+		if err != nil {
+			return err
+		}
+		if req == nil {
```

Comment:
For the case that the request body is empty (e.g., MultipartType_ControllerStatus, MultipartType_MeterFeatures, MultipartType_TableDesc), `req` should be nil, then this error is not expected, correct?

Reply:
That is right. I will fix this.